openfisca_core/entities.py

```diff
@@ -463,11 +512,25 @@ def value_from_person(self, array, role, default = 0):
 
         return result
 
-    def value_from_first_person(self, array):
+    def value_kth_person(self, k, array, default = 0):
+        """
+            Get the value of criteria for the kth persons of each entity.
+            If the kth person does not exist, get default instead
+            The result has the same dimension than the number of entities
+        """
         self.simulation.persons.check_array_compatible_with_entity(array)
-        position_filter = (self.members_position == 0)
+        positions = self.members_position
+        nb = self.nb_persons()
+        mask = self.ordered_members_mask
```

Comment:
Why is it called a mask ? it sounds to me as a map and I don't understand the mask metaphore 😕

Reply:
Yes, you're right, it's not a mask, a mask is only when the vector is a boolean and that you "mask" the values for which the boolean is false.

Ok with renaming to map.